controllers/storagecluster/noobaa_system_reconciler.go

```diff
@@ -201,8 +201,8 @@ func (r *StorageClusterReconciler) setNooBaaDesiredState(nb *nbv1.NooBaa, sc *oc
 		if sc.Spec.MultiCloudGateway.ExternalPgConfig != nil && sc.Spec.MultiCloudGateway.ExternalPgConfig.PGSecretName != "" {
 			nb.Spec.ExternalPgSecret = &corev1.SecretReference{Name: sc.Spec.MultiCloudGateway.ExternalPgConfig.PGSecretName, Namespace: sc.Namespace}
 			nb.Spec.ExternalPgSSLUnauthorized = sc.Spec.MultiCloudGateway.ExternalPgConfig.AllowSelfSignedCerts
+			nb.Spec.ExternalPgSSLRequired = sc.Spec.MultiCloudGateway.ExternalPgConfig.EnableTLS
 			if sc.Spec.MultiCloudGateway.ExternalPgConfig.TLSSecretName != "" {
```

Comment:
What is the expected behavior when,

1. `enableTLS: false` but user provides `tlsSecretName`?
2. `enableTLS: true`, `allowSelfSignedCerts: false`, `tlsSecretName` is empty?
3. `enableTLS: true`, `allowSelfSignedCerts: true`, `tlsSecretName` is not empty?

Reply:
I think this can answer these questions: https://github.com/noobaa/noobaa-operator/blob/master/doc/external-postgres.md#ssl-support
1. externalPgSSLRequired and externalPgSSLUnauthorized are used for the server side, the 1st will force using ssl when talking to the external server, the 2nd will let the user connect to the server even if the client doesn't trust the CA.
2. externalPgSSLSecret will be used for the client side and will be used if the server wants the clients to connect with certificate instead of user/password

I do think that it's not allowed to use client-side secret without enabling TLS, but other than that the rest are fine. 
Hopes that clears it.

Reply:
> 
> I do think that it's not allowed to use client-side secret without enabling TLS, but other than that the rest are fine. Hopes that clears it.
>
@vbnrh We should add a validation for this case.

Reply:
Added validation in the If statement.

Thanks @jackyalbo for the clarification.